AzureAD/microsoft-authentication-library-for-android
# Token fetch stress testing.

## What does this PR do?

- Stress testing `acquireTokenSilent` with cached token.

## Description of task to be completed.

- Create MSAL application for implementation of stress testing.
- Add tests for `acquireTokenSilent` using 10 threads calling `acquireTokenSilent` with cached tokens for 6 hours.

## Provide any relevant context.
- There is existing documentation on ADAL related to stress testing on this commit https://github.com/AzureAD/azure-activedirectory-library-for-android/commit/21db2f78875580446dcd8748d375115c849a0e7e 

## How can this be manually tested?
- Checkout to the `git/stress-test` branch on MSAL
- Update `android-complete`'s settings.gradle with the following

```
:include(":msalstresstests")
project(":msalstresstests").projectDir = new File("msal/testapps/stresstests")
```

- From android studio, run the `msalstresstests` on your physical device/emulator
- Go to the `acquire token silent` fragment (Should be the first selected on the navigation drawer)
- Click on the button to run the tests. (The tests will run for 6 hours)

## Design
https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/git/stress-test/testapps/stresstests/README.md
